src/wallet/scriptpubkeyman.cpp

```diff
@@ -842,7 +895,14 @@ void LegacyScriptPubKeyMan::SetHDChain(const CHDChain& chain, bool memonly)
     if (!memonly && !WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain))
         throw std::runtime_error(std::string(__func__) + ": writing chain failed");
 
-    hdChain = chain;
+    m_hd_chain = chain;
+}
+
+void LegacyScriptPubKeyMan::AddInactiveHDChain(const CHDChain& chain)
```

Comment:
Per today's review club session, should `AddInactiveHDChain` be called not only from `LoadWallet` but also from sethdseed (or elsewhere), and tests updated to check for that case.

Reply:
Added a call to `AddInactiveHDChain` to `SetHDCHain` which is where chain rotation occurs.